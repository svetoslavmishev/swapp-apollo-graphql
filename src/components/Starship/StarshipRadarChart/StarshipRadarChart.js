import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import RadarChart from 'react-svg-radar-chart';
import 'react-svg-radar-chart/build/css/index.css';

import { ThemeContext } from '../../../themeContext';
import compareValues from '../../../helpers';
import styles from './StarshipRadarChartStyles';

const StarshipRadarChart = ({
  starship,
  starshipsByClass: {
    allStarships: { edges }
  }
}) => {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });

  const data = [
    {
      data: {
        maxAtmosphericSpeed: compareValues(
          edges,
          starship.maxAtmosphericSpeed,
          'maxAtmosphericSpeed'
        ),
        cost: compareValues(edges, starship.cost, 'cost'),
        maxMLPerHour: compareValues(
          edges,
          starship.maxMLPerHour,
          'maxMLPerHour'
        ),
        crew: compareValues(edges, starship.crew, 'crew'),
        hyperdriveRating: compareValues(
          edges,
          starship.hyperdriveRating,
          'hyperdriveRating'
        )
      },
      meta: { color: '#4bd5ee' }
    }
  ];

  return (
    <div className="md:ml-20">
      <RadarChart
        options={{
          dots: true,
          dotProps: () => ({
            className: `dot ${classes.dots}`
          }),
          axes: false,
          scales: 5,
          scaleProps: () => ({
            className: 'axis',
            fill: 'none'
          }),
          zoomDistance: 1.3,
          captionProps: () => ({
            className: classes.customCaption
          })
        }}
        captions={{
          maxAtmosphericSpeed: 'Max Atm. Speed',
          maxMLPerHour: 'Max ML/h',
          hyperdriveRating: 'HyperD Rat.',
          crew: 'Crew',
          cost: 'Cost'
        }}
        data={data}
        size={350}
      />
    </div>
  );
};

StarshipRadarChart.propTypes = {
  starshipsByClass: PropTypes.object
};

export default StarshipRadarChart;
