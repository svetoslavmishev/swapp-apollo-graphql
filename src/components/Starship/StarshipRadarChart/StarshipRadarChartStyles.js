import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  root: {
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  },
  customCaption: {
    fill: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor,
    textAnchor: 'middle',
    fontSize: 14,
    fontWeight: 600
  },
  dots: {
    fill: theme.palette['light'].PrimaryHeading.fontColor,
    r: 3
  }
}));

export default styles;
