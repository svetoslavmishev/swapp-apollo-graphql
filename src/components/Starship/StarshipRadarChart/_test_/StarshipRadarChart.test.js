import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import StarshipRadarChart from '../StarshipRadarChart';

describe('<StarshipRadarChart />', () => {
  it('renders <StarshipRadarChart /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <StarshipRadarChart />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
