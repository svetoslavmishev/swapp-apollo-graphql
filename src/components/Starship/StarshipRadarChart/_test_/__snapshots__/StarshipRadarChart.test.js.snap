// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarshipRadarChart /> renders <StarshipRadarChart /> without error 1`] = `ShallowWrapper {}`;
